source/ada/lsp-ada_documents.ads

```diff
@@ -63,6 +63,12 @@ package LSP.Ada_Documents is
    function Versioned_Identifier
      (Self : Document) return LSP.Messages.VersionedTextDocumentIdentifier;
 
+   function Unit
+     (Self    : Document;
+      Context : LSP.Ada_Contexts.Context)
+      return Libadalang.Analysis.Analysis_Unit;
+   --  Return the analysis unit for Self in the given context
```

Comment:
Can we move part of `On_ALS_Show_Dependencies_Request` to `lsp-ada_documents.ads`  instead of make `Unit` visible?

Reply:
Having a way to retrieve the Unit of a Document seems a good idea to me, why you don't want to make it visible?

Reply:
`Document` type provides encapsulation over `Analysis_Unit`. Make `Unit` visible breaks this encapsulation. As result more logic concentrates in `Ada_Handler`, but `Ada_Handler` is expected to by just a protocol adapted.

General architecture of Ada Language server (that is located in `source/ada` directory) consists of:

- `Context_Set` - is set of context, that represents aggregated project. It owns `Context` objects.
- `Context` - a context represents non aggregate project.
- `Document` - a representation of Ada text file. It could be interpreted in several `Context`s if the corresponding Ada file belongs to several projects of an aggregated project.
- `Ada_Handlers.Message_Handler` - handler just a protocol adapter, that uses `Document`, `Context` and `Context_Set` to execute LSP queries.

Put all logic into the `Message_Handler` will result in a mess of code in one huge compilation unit. So, moving code into `Document` helps us to have more clean design.

If a new functionality fits into this architecture, then i would prefer not to make `Unit` visible. Otherwise we should find a new proper architecture.

Reply:
Understood. I tried to follow what you are saying in my last patch.